sysdig/internal/client/secure/scanningpolicies.go

```diff
@@ -0,0 +1,146 @@
+package secure
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+)
+
+func (client *sysdigSecureClient) scanningPoliciesURL() string {
+	return fmt.Sprintf("%s/api/scanning/v1/policies", client.URL)
+}
+
+func (client *sysdigSecureClient) scanningPolicyAssignmentURL() string {
+	return fmt.Sprintf("%s/api/scanning/v1/mappings?bundleId=default", client.URL)
+}
+
+func (client *sysdigSecureClient) scanningPolicyURL(scanningPolicyId string) string {
+	return fmt.Sprintf("%s/api/scanning/v1/policies/%s", client.URL, scanningPolicyId)
+}
+
+// Scanning Policies
+
+func (client *sysdigSecureClient) CreateScanningPolicy(ctx context.Context, scanningPolicyRequest ScanningPolicy) (scanningPolicy ScanningPolicy, err error) {
+	response, err := client.doSysdigSecureRequest(ctx, http.MethodPost, client.scanningPoliciesURL(), scanningPolicyRequest.ToJSON())
+	if err != nil {
+		return
+	}
+	defer response.Body.Close()
+
+	if response.StatusCode != http.StatusOK {
+		err = errorFromResponse(response)
+		return
+	}
+
+	body, err := io.ReadAll(response.Body)
+	if err != nil {
+		return
+	}
+
+	return ScanningPolicyFromJSON(body), nil
+}
+
+func (client *sysdigSecureClient) GetScanningPolicyById(ctx context.Context, scanningPolicyID string) (scanningPolicy ScanningPolicy, err error) {
+	response, err := client.doSysdigSecureRequest(ctx, http.MethodGet, client.scanningPolicyURL(scanningPolicyID), nil)
+	if err != nil {
+		return
+	}
+	defer response.Body.Close()
+
+	if response.StatusCode != http.StatusOK {
+		return ScanningPolicy{}, errorFromResponse(response)
+	}
+
+	body, err := io.ReadAll(response.Body)
+	if err != nil {
+		return
+	}
+	return ScanningPolicyFromJSON(body), nil
+}
+
+func (client *sysdigSecureClient) UpdateScanningPolicyById(ctx context.Context, scanningPolicyRequest ScanningPolicy) (scanningPolicy ScanningPolicy, err error) {
+	response, err := client.doSysdigSecureRequest(ctx, http.MethodPut, client.scanningPolicyURL(scanningPolicyRequest.ID), scanningPolicyRequest.ToJSON())
+	if err != nil {
+		return
+	}
+	defer response.Body.Close()
+
+	if response.StatusCode != http.StatusOK {
+		return ScanningPolicy{}, errorFromResponse(response)
+	}
+
+	body, err := io.ReadAll(response.Body)
+	if err != nil {
+		return
+	}
+	return ScanningPolicyFromJSON(body), nil
+}
+
+func (client *sysdigSecureClient) DeleteScanningPolicyById(ctx context.Context, scanningPolicyID string) error {
+	response, err := client.doSysdigSecureRequest(ctx, http.MethodDelete, client.scanningPolicyURL(scanningPolicyID), nil)
+	if err != nil {
+		return err
+	}
+	defer response.Body.Close()
+
+	if response.StatusCode != http.StatusNoContent && response.StatusCode != http.StatusOK {
```

Comment:
Do we have this condition here because we do not know what is the 2xx respond code of the DELETE action or server could return either of those two http codes?

Reply:
Per this [test](https://github.com/draios/secure-backend/blob/main/scanning/legacy/backend/test/api/integration/test/policies_test.go#L83) for that same endpoint in our backend we check the `StatusNoContent`

But, checking [backend code](https://github.com/draios/secure-backend/blob/main/scanning/legacy/backend/api/api.go#L1212) seems we send the `StatusNoContent` as well as result